sale_coupon_product_management/models/sale_coupon_program_option.py

```diff
@@ -98,7 +98,14 @@ def _get_related_record(self, program, path):
             return program
         return operator.attrgetter(path)(program)
 
-    def get_program_values(self):
+    def _update_values_from_program(self, opt_vals, program=None):
```

Comment:
Since that method does nothing when program isn't set, why do you set this attribute as optional ?

Reply:
Because it comes from parent method, which has that record. Though I guess I could move check outside that function, to look more clean.